test/Generics/function_defs.swift

```diff
@@ -295,5 +295,16 @@ func badTypeConformance1<T>(_: T) where Int : EqualComparable {} // expected-err
 
 func badTypeConformance2<T>(_: T) where T.Blarg : EqualComparable { } // expected-error{{'Blarg' is not a member type of 'T'}}
 
+func badTypeConformance3<T>(_: T) where (T) -> () : EqualComparable { }
+// expected-error@-1{{type '(T) -> ()' in conformance requirement does not refer to a generic parameter or associated type}}
+
+// FIXME: The first error is emitted twice.
```

Comment:
The error is emitted twice here, but that's a separate problem, probably in `GenericSignatureBuilder`; it happens with the following example as well. 
```
func foo<T>(...) where T == @escaping (Int) -> ()
```

Reply:
Yeah, that's because we walk the parameters and requirements twice when validating generic functions. It's not a result of your change.

Reply:
@slavapestov Should I add FIXMEs for duplicate errors, or it isn't considered an issue here?